tuya.md

```diff
@@ -0,0 +1,283 @@
+# Supporting Tuya Devices
+
+Caution, the following should work for most Tuya devices, some devices may require additional reverse engineering to unlock all functions.
+
+# Identify Tuya Data Points
+
+The first step in building a Tuya quirk is to identify the Tuya Datapoints (DPs) for the device. There are two ways ways to do this.
+
+1. If the device is supported by Zigbee2MQTT, the DPs can be captured from the [herdsman converter](https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/src/devices/tuya.ts).
+2. Using a Tuya hub, the DPs can be captured from the Tuya developer's console. See [Zigbee2MQTT Documentation](https://www.zigbee2mqtt.io/advanced/support-new-devices/03_find_tuya_data_points.html)
+
+# Using the datapoints to develop a Tuya Quirk
+
+Once the DPs are identified, the quirk can be built. For each DP, identify the correct replacement for the quirk using the available methods below. For commonly used replacements, such as a power configuration cluster, we can use a convenience method, such as `.tuya_battery`. 
+
+Note: Convenience methods will only work once, if your device has multiple identical clusters, such as on_off, use multiple `.tuya_switch` calls instead. Otherwise, only one switch will be generated.
+
+For more complex replacements you may need to use a lower level method, such as `.tuya_dp_attribute` or even `.tuya_dp` and `.tuya_attribute`.
+
+All V2 QuirkBuilder methods are available, so using `.tuya_dp` to add a DP converter then `.adds` to add the correct class is valid.
+
+Most V2 quirks will match only on the model and manufacturer, this reduces duplicated code where a new variant appears with a slightly different signature. Should you need to filter on a signature also, use `.filter`.
+
+```python
+from zigpy.quirks import signature_matches
+
+.filter(signature_matches(device_signature))
+```
+
+Once the quirk is complete, enable custom quirks and test. See [Configuration - YAML in ZHA documentation](https://www.home-assistant.io/integrations/zha/).
```

Comment:
We should probably add a small snippet here or to our quirks wiki (and link it here) for what's needed to activate custom quirks.